apps/api-extractor/src/generators/dtsRollup/PackageMetadataManager.ts

```diff
@@ -26,37 +33,64 @@ export class PackageMetadata {
    */
   public readonly aedocSupported: boolean;
 
-  private readonly _packageJsonLookup: PackageJsonLookup;
-
-  public constructor(packageJsonPath: string, packageJsonLookup: PackageJsonLookup) {
-    this._packageJsonLookup = packageJsonLookup;
+  public constructor(packageJsonPath: string, packageJson: IPackageJson, aedocSupported: boolean) {
     this.packageJsonPath = packageJsonPath;
-
-    this.packageJson = this._packageJsonLookup.loadPackageJson(packageJsonPath);
-
-    this.aedocSupported = false;
-
-    if (this.packageJson.tsdoc) {
-      if (this.packageJson.tsdoc.tsdocFlavor) {
-        if (this.packageJson.tsdoc.tsdocFlavor.toUpperCase() === 'AEDOC') {
-          this.aedocSupported = true;
-        }
-      }
-    }
+    this.packageJson = packageJson;
+    this.aedocSupported = aedocSupported;
   }
 }
 
 /**
  * This class maintains a cache of analyzed information obtained from package.json
  * files.  It is built on top of the PackageJsonLookup class.
+ *
+ * @remarks
+ *
+ * IMPORTANT: Don't use PackageMetadataManager to analyze source files from the current project:
+ * 1. Files such as tsdoc-metadata.json may not have been built yet, and thus may contain incorrect information.
+ * 2. The current project is not guaranteed to have a package.json file at all.  For example, API Extractor can
+ *    be invoked on a bare .d.ts file.
+ *
+ * Use ts.program.isSourceFileFromExternalLibrary() to test source files before passing the to PackageMetadataManager.
  */
 export class PackageMetadataManager {
+  public static tsdocMetadataFilename: string = 'tsdoc-metadata.json';
+
   private readonly _packageJsonLookup: PackageJsonLookup;
+  private readonly _logger: ILogger;
   private readonly _packageMetadataByPackageJsonPath: Map<string, PackageMetadata>
     = new Map<string, PackageMetadata>();
 
-  public constructor(packageJsonLookup: PackageJsonLookup) {
+  public static writeTsdocMetadataFile(packageJsonFolder: string): void {
+    // This feature is still being standardized: https://github.com/Microsoft/tsdoc/issues/7
+    // In the future we will use the @microsoft/tsdoc library to read this file.
+    const tsdocMetadataPath: string = path.join(packageJsonFolder,
+      'dist', PackageMetadataManager.tsdocMetadataFilename);
```

Comment:
The rollup file may not exist.  But we could put it in the same directory as the default entry point for the NPM package.

---
In reply to: [236576554](https://github.com/Microsoft/web-build-tools/pull/960#discussion_r236576554) [](ancestors = 236576554)

Reply:
I've updated [TSDoc #7](https://github.com/Microsoft/tsdoc/issues/7) to reflect this idea.

However I'm not going to implement it here, since this is a temporary workaround.

---
In reply to: [236597139](https://github.com/Microsoft/web-build-tools/pull/960#discussion_r236597139) [](ancestors = 236597139,236576554)